src/main/java/edu/kit/compiler/optimizations/NodeCosts.java

```diff
@@ -0,0 +1,70 @@
+package edu.kit.compiler.optimizations;
+
+import firm.bindings.binding_irnode.ir_opcode;
+import firm.nodes.Node;
+
+/**
+ * This class contains an estimate of the cost (in abstract CPU cycles) of Firm
+ * nodes.
+ */
+public final class NodeCosts {
+
+    private NodeCosts() { }
+
+    /**
+     * Get an estimate of the cost of the given node without considering
+     * surrounding nodes. More advanced instruction selection is not taken into
+     * account.
+     */
+    public static int getCost(Node node) {
+        return getCost(node.getOpCode());
+    }
+
+    /**
+     * Get an estimate of the cost of a node with the given opcode without
+     * considering additional information for that node.
+     */
+    public static int getCost(ir_opcode opcode) {
+        switch (opcode) {
+            case iro_Const:
+                return 0;
+            case iro_Address:
+                return 0;
+            case iro_Proj:
+                return 0;
+            case iro_Tuple:
+                return 0;
+            case iro_NoMem:
+                return 0;
+            case iro_Mul:
+                return 4;
+            case iro_Mulh:
+                return 4;
+            case iro_Div:
+                return 25;
+            case iro_Mod:
+                return 25;
+            case iro_Alloc:
+                return 50;
```

Comment:
Alloc usually involves multiple loads, and would thus probably be even more expensive